### DISCUSSION TITLE
Can't compile CSS using Next.js 15 with turbo

### DISCUSSION THREAD

**Discussion by doei** [tailwindlabs/tailwindcss#14751] [Help] [answered]
After trying to upgrade to next 15, my css compilation does not seem to work anymore, I get this error:

```
 ⨯ ./src/styles/globals.css:902:16
Parsing css source code failed
  900 | }
  901 | .min-h-\[\!\] {
> 902 |   min-height: !;
      |                ^
  903 | }
  904 | .min-h-\[40\%\] {
  905 |   min-height: 40%;

Unexpected token Delim('!') at [project]/src/styles/globals.css:901:15
```

I can't figure out what is happening, I've tried removing my tailwind plugins but that didn't seem to help...

**Comment by tobi1220** [comment, marked as answer]
> `min-height: !;`

Is this your CSS? If not, you might be getting the same error I was facing, which was because I was storing a js file (in my case prism.js) in one of the directories specified for `content` in the tailwind config, and it contained a string that would be captured as a utility class (and added to my globals.css by the `@tailwind utilities` directive), which broke the CSS.

Check `content` in your tailwind.config.css and see if there might be any suspicious file.

**Reply by philipp-spiess** [reply]
Yeah this particular line of CSS does suggest that you have the string `min-h-[!]` somewhere in your templates.

**Reply by doei** [reply]
Thank you guys so much for the hint, I don't know how I did not think of this...

This led me to one buried tailwind class that indeed had `min-h-[!]`, and indeed in one of the directories specified by `content` in the tailwind config.

Removing it fixed the error, but what I don't understand is how this class was functional before, the element's styling was definitely affected.

The original line read 'p-4 w-[!] min-h-[!] max-w-[!] items-center'` I don't how this was supposed to work, the intent was to have a container element fit its content. I cannot find any reference to this syntax though, and I cannot remember where I originally got that from...

Regardless, thanks for the help.

**Reply by tobi1220** [reply]
This is just a shot in the dark, but I could imagine this is due to either turbopack itself or tailwind in turbopack processing CSS differently. I faced the exact same behavior with a very similar error message that would only show with `--turbo`. I guess the broken styles were there before and just didn't do anything, let alone throw an error, or the broken styles were just filtered out before.
